test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.Protocol.cs

```diff
@@ -565,6 +565,62 @@ public class Protocol
                     await connection.DisposeAsync().OrTimeout();
                 }
             }
+
+            [Fact]
+            public async Task ClientPingsAutomatically()
+            {
+                var connection = new TestConnection();
+                var hubConnection = CreateHubConnection(connection);
+
+                hubConnection.TickRate = TimeSpan.FromMilliseconds(30);
+                hubConnection.PingInterval = TimeSpan.FromMilliseconds(80);
+
+                try
+                {
+                    await hubConnection.StartAsync().OrTimeout();
+                    await Task.Delay(120);
+
+                    var received = await connection.ReadSentTextMessageAsync().OrTimeout(TimeSpan.FromMilliseconds(100));
+                    Assert.Equal("{\"type\":6}", received);
+                }
+                finally
+                {
+                    await hubConnection.DisposeAsync().OrTimeout();
+                    await connection.DisposeAsync().OrTimeout();
+                }
+            }
+
+            [Fact]
+            public async Task ClientPingsMultipleTimes()
+            {
+                var connection = new TestConnection();
+                var hubConnection = CreateHubConnection(connection);
+
+                hubConnection.TickRate = TimeSpan.FromMilliseconds(30);
+                hubConnection.PingInterval = TimeSpan.FromMilliseconds(80);
+
+                try
+                {
+                    await hubConnection.StartAsync().OrTimeout();
+                    await Task.Delay(220);
+
+                    var firstPing = await connection.ReadSentTextMessageAsync().OrTimeout(TimeSpan.FromMilliseconds(5));
+                    Assert.Equal("{\"type\":6}", firstPing);
+
+                    var secondPing = await connection.ReadSentTextMessageAsync().OrTimeout(TimeSpan.FromMilliseconds(5));
+                    Assert.Equal("{\"type\":6}", secondPing);
+                    
+                    // with a 200ms delay, we should have enough time for 2 pings but not 3
+                    await Assert.ThrowsAsync<TimeoutException>( 
```

Comment:
Ah, this is what's causing the test failure. The problem is that `Task.Delay` doesn't guarantee that you'll run after exactly 220ms, it just means that you'll go back on the scheduler queue after 220ms. On slower machines (like our CI machines), when a lot of other tests are running in parallel, it might take longer than 220ms, so you may end up with more than 2 pings.

I don't think we need this assert anyway, the goal is to test that we get at least 2 pings, we're OK with getting more.